src/base/tristatebool.h

```diff
@@ -36,15 +36,26 @@ class TriStateBool
     static const TriStateBool False;
     static const TriStateBool True;
 
-    TriStateBool() = default;
-    TriStateBool(const TriStateBool &other) = default;
+    constexpr TriStateBool() = default;
+    constexpr TriStateBool(const TriStateBool &other) = default;
+    explicit TriStateBool(int value);  // add constexpr when using C++14
 
-    explicit TriStateBool(int value);
-    explicit operator int() const;
+    explicit constexpr operator int() const
```

Comment:
Why did you move definitions of constexpr methods into header? Is it required?

Reply:
Yes, constexpr requires the function definition be available at compile-time.